server/src/main/java/org/apache/cassandra/sidecar/server/Server.java

```diff
@@ -167,9 +167,7 @@ public Future<CompositeFuture> close()
         periodicTaskExecutor.close(periodicTaskExecutorPromise);
         closingFutures.add(periodicTaskExecutorPromise.future());
 
-        Promise<Void> closeSidecarClientPromise = Promise.promise();
-        sidecarClientProvider.close(closeSidecarClientPromise);
-        closingFutures.add(closeSidecarClientPromise.future());
+        sidecarClientProvider.close();
```

Comment:
Still need to add it in the future chain. 

```suggestion
        closingFutures.add(Future.future(p -> {
            sidecarClientProvider.close();
            p.complete();
        }));
```